src/v0/destinations/rudder_test/networkHandler.js

```
@@ -0,0 +1,92 @@
+    throw new NetworkError(
+      `Request failed for ${DESTINATION} with status: ${status}`,
+      status,
+      {
+        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(status),
+      },
+      destinationResponse,
+    );
+  }
+
+  // Simulate errors based on test behavior using existing statusCode and errorMessage
+  if (testBehavior && testBehavior.statusCode && testBehavior.statusCode !== 200) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)